op-service/dial/active_l2_provider_test.go

```diff
@@ -0,0 +1,468 @@
+
+// newActiveL2EndpointProvider constructs a new ActiveL2RollupProvider using the test harness setup.
+func (et *endpointProviderTest) newActiveL2RollupProvider(checkDuration time.Duration) (*ActiveL2RollupProvider, error) {
+	rollupDialer := func(ctx context.Context, timeout time.Duration, log log.Logger, url string) (RollupClientInterface, error) {
+		for i, client := range et.rollupClients {
+			if url == fmt.Sprintf("rollup%d", i) {
+				return client, nil
+			}
+		}
+		return nil, fmt.Errorf("unknown test url: %s", url)
+	}
+
+	// make the "URLs"
+	rollupUrls := make([]string, len(et.rollupClients))
+	for i := range et.rollupClients {
+		rollupUrl := fmt.Sprintf("rollup%d", i)
+		rollupUrls[i] = rollupUrl
+	}
+
+	return newActiveL2RollupProvider(
+		context.Background(),
+		rollupUrls,
+		checkDuration,
+		1*time.Minute,
+		testlog.Logger(et.t, log.LvlDebug),
+		rollupDialer,
+	)
+}
+
+// newActiveL2EndpointProvider constructs a new ActiveL2EndpointProvider using the test harness setup.
+func (et *endpointProviderTest) newActiveL2EndpointProvider(checkDuration time.Duration) (*ActiveL2EndpointProvider, error) {
+	mockRollupDialer := func(ctx context.Context, timeout time.Duration, log log.Logger, url string) (RollupClientInterface, error) {
+		for i, client := range et.rollupClients {
+			if url == fmt.Sprintf("rollup%d", i) {
+				return client, nil
+			}
+		}
+		return nil, fmt.Errorf("unknown test url: %s", url)
+	}
+
+	mockEthDialer := func(ctx context.Context, timeout time.Duration, log log.Logger, url string) (EthClientInterface, error) {
+		for i, client := range et.ethClients {
+			if url == fmt.Sprintf("eth%d", i) {
+				return client, nil
+			}
+		}
+		return nil, fmt.Errorf("unknown test url: %s", url)
+	}
+
+	// make the "URLs"
+	rollupUrls := make([]string, len(et.rollupClients))
+	for i := range et.rollupClients {
+		rollupUrl := fmt.Sprintf("rollup%d", i)
+		rollupUrls[i] = rollupUrl
+	}
+	ethUrls := make([]string, len(et.ethClients))
+	for i := range et.ethClients {
+		ethUrl := fmt.Sprintf("eth%d", i)
+		ethUrls[i] = ethUrl
+	}
+
+	return newActiveL2EndpointProvider(
+		context.Background(),
+		ethUrls,
+		rollupUrls,
+		checkDuration,
+		1*time.Minute,
+		testlog.Logger(et.t, log.LvlDebug),
+		mockEthDialer,
+		mockRollupDialer,
+	)
+}
+
+// TestRollupProvider_FailoverOnInactiveSequencer verifies that the ActiveL2RollupProvider
+// will switch to the next provider if the current one becomes inactive.
+func TestRollupProvider_FailoverOnInactiveSequencer(t *testing.T) {
+	ept := setupEndpointProviderTest(t, 2)
+	primarySequencer, secondarySequencer := ept.rollupClients[0], ept.rollupClients[1]
+
+	primarySequencer.ExpectSequencerActive(true, nil) // respond true once on creation
+	primarySequencer.ExpectSequencerActive(true, nil) // respond true again when the test calls `RollupClient()` the first time
+
+	shortCheckDuration := 1 * time.Microsecond
+	rollupProvider, err := ept.newActiveL2RollupProvider(shortCheckDuration)
```

Comment:
if you want to just always have it check in tests, I think you can just use `0` duration
```suggestion
	rollupProvider, err := ept.newActiveL2RollupProvider(0)
```

you could also add to the doc of the ActiveRollupProvider that 0 enables check on every call to `RollupClient`.